Add layer specific disclaimers

RobQuincey-DC opened the pull request:
This PR adds optional disclaimers to individual layers.

Disclaimers appear the first time a user switches a layer on, and then reappears depending on the frequency. They are designed to warn users about terms of use or issues with a particular layer. 

## How it works
You create a 'disclaimer' in the management console, and then in the layer details page you choose the disclaimer you want.

- A layer can only have one disclaimer.
- A disclaimer can be tied to multiple layers
  - In this case, the disclaimer will only appear when selecting the first layer. Any subsequent layers with the same disclaimer will not show, allowing you to 'bundle' layers together with the same disclaimer and prevent having to show users the same disclaimer constantly
- The maximum amount a layer disclaimer can be shown is once per 'session' (defined here - https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage).
- Disclaimers can be written in HTML and will appear in a modal
- Disclaimers can have custom button text, or use the default of 'Close'
- Disclaimers have a frequency that works the same as the welcome message, with the slight difference being the 'always' option will only show once per session.

## Limitations
If a layer is already turned on (either through layer customisations or via a share link) the disclaimer will not trigger unless the user turns the layer off and back on. This is to prevent being potentially bombarded with disclaimers on first load. Imagine you have a  welcome message, a tour, a cookie control banner and three layers turned on with separate disclaimers, the user would have to dismiss SIX separate overlays before the map could be used.
We could improve the logic to bundle any disclaimers into the welcome screen in one go, but this was deemed out of scope for initial implementation.

paul-dorsetcouncil reviewed (changes requested):
I like the implementation in general but I have two comments, one is a nit pick and the other might need looking at before we approve? 

1. Could the bubble appear more centrally? I think the tour is in the middle but disclaimer is appearing at the top slightly over the header. This is not really an issue and happy to go with the majority if we are happy with the location.
2. It appears that the bubble clears if you click anywhere outside the popup. This is fine for some but I think for the "Accept" example we will need to prevent that behaviour and force the user to actually click the button. This one I think we might want to investigate further. 

Other than these suggestions, I think it looks good and happy to accept after we've discussed point 2 above.

RobQuincey-DC commented:
@paul-dorsetcouncil Good spot, that was supposed to not be dismissible by clicking outside. That is now fixed so you have to click the button to dismiss the modal. I've also moved the modal to the middle of the screen.

paul-dorsetcouncil reviewed (approved):
Looks good, thank you.
